src/Agent.Plugins/PipelineArtifact/PipelineArtifactServer.cs

```diff
@@ -66,7 +76,14 @@ public class PipelineArtifactServer
                                                                         propertiesDictionary, 
                                                                         cancellationToken);
 
-                context.Output(StringUtil.Loc("AssociateArtifactWithBuild", artifact.Id, pipelineId));
+                    },
+                    maxRetries: 3,
```

Comment:
A higher number for retires might be better ?

Reply:
This is retry on top of retry. I don't think we want to retry to many times.  And the timeout 300 secs should give the server sufficient time to complete the process.